xchange-stream-btcmarkets/src/main/java/info/bitrich/xchangestream/btcmarkets/BTCMarketsStreamingAdapters.java

```diff
@@ -41,4 +52,49 @@ public static OrderBook adaptOrderbookMessageToOrderbook(
             .map((o) -> toLimitOrder.apply(o, Order.OrderType.BID))
             .collect(Collectors.toList()));
   }
+
+  public static Ticker adaptTickerMessageToTicker(BTCMarketsWebSocketTickerMessage message)
+      throws InvalidFormatException {
+    CurrencyPair currencyPair = adaptMarketIdToCurrencyPair(message.getMarketId());
+
+    return new Ticker.Builder()
+        .instrument(currencyPair)
+        .last(message.getLastPrice())
+        .bid(message.getBestBid())
+        .ask(message.getBestAsk())
+        .timestamp(DateUtils.fromISODateString(message.getTimestamp()))
+        .volume(message.getVolume24h())
+        .build();
+  }
+
+  public static Trade adaptTradeMessageToTrade(BTCMarketsWebSocketTradeMessage message)
+      throws InvalidFormatException {
+    CurrencyPair currencyPair = adaptMarketIdToCurrencyPair(message.getMarketId());
+    return new Trade.Builder()
+        .instrument(currencyPair)
+        .id(message.getTradeId())
+        .price(message.getPrice())
+        .timestamp(DateUtils.fromISODateString(message.getTimestamp()))
+        .type(getOrderType(message.getSide()))
+        .build();
+  }
+
+  private static OrderType getOrderType(String side) {
+    OrderType orderType = null;
+    switch (side) {
+      case "Ask":
+        orderType = OrderType.ASK;
+      case "Bid":
+        orderType = OrderType.BID;
+      default:
+        LOG.info("Trade side {} not recognised, set to null", side);
+    }
+    return orderType;
+  }
+
+  public static OrderBook adaptOrderUpdateMessageToOrderBook(
```

Comment:
since this isn't required given this class is basically static functions no need to have this function here.

Reply:
why do we have this method?

Reply:
This method was updated, it is just a convinience method to use inline, the exchange trade message used "Ask" and "Bid" or "ask" and "bid" randomnly. The orderType is an enumerator so need to match them. Anyway, check the new update once pulled.